### ISSUE TITLE
"Parent has no child drawable!" on Nokia's old devices if scaleType is specified for failure drawable

### ISSUE THREAD

**Issue by Kazemir** [facebook/fresco#2241]
### Description

We've got some strange crashes on old Nokia's devices on `setFailureImage(resourceId, scaleType)` method call:
```
Non-fatal Exception: java.lang.NullPointerException: Parent has no child drawable!
       at com.facebook.common.internal.Preconditions.checkNotNull(Preconditions.java:226)
       at com.facebook.drawee.generic.WrappingUtils.wrapChildWithScaleType(WrappingUtils.java:129)
       at com.facebook.drawee.generic.GenericDraweeHierarchy.getScaleTypeDrawableAtIndex(GenericDraweeHierarchy.java:361)
       at com.facebook.drawee.generic.GenericDraweeHierarchy.setFailureImage(GenericDraweeHierarchy.java:465)
       at com.facebook.drawee.generic.GenericDraweeHierarchy.setFailureImage(GenericDraweeHierarchy.java:484)
       at my.favorite.app.PhotoPreviewBinder.onBindViewHolder(PhotoPreviewBinder.java:37)
       ...
```

### Reproduction

I don't know how to reproduce this crash, because we don't have that device. But I can show the lines of code where crash is occurred and how we handle it:
```
SimpleDraweeView photo = findViewById(R.id.photo);

ImageRequest imageRequest = ImageRequestBuilder.newBuilderWithSource(Uri.parse(entry.url))
	.setResizeOptions(ResizeOptions.forSquareSize(200))
	.build();
photo.setImageRequest(imageRequest);

final GenericDraweeHierarchy hierarchy = photo.getHierarchy();
try {
	// **Crash is here!**
	hierarchy.setFailureImage(R.drawable.video_preview_stub, ScalingUtils.ScaleType.CENTER_CROP);
} catch (Exception e) {
	// And no crash here
	hierarchy.setFailureImage(R.drawable.video_preview_stub);
	ExceptionHook.hook(e);
}
```

### Additional Information

* Fresco version: 1.11.0
* Devices: Nokia_X (RM-980), Nokia_XL (RM-1030), NokiaX2DS (RM-1013)
* Android version: Android 4.1.2 (API 16), Android 4.3 (API 18)
* Arch: armv7l

**Comment by iadeelzafar** [comment]
@Kazemir is this issue still reproducible?

 If yes then @erikandre what do you think of the proposed solution in the PR description? Should I go further and implement this?

**Comment by Kazemir** [comment]
It looks like bug was fixed in 2.0.0

**Comment by YvesCheung** [comment]
```
java.lang.NullPointerException: Parent has no child drawable!
	at com.facebook.common.internal.Preconditions.checkNotNull(Preconditions.java:231)
	at com.facebook.drawee.generic.WrappingUtils.wrapChildWithScaleType(WrappingUtils.java:126)
	at com.facebook.drawee.generic.GenericDraweeHierarchy.getScaleTypeDrawableAtIndex(GenericDraweeHierarchy.java:360)
	at com.facebook.drawee.generic.GenericDraweeHierarchy.setFailureImage(GenericDraweeHierarchy.java:460)
	at com.facebook.drawee.generic.GenericDraweeHierarchy.setFailureImage(GenericDraweeHierarchy.java:479)
	at ryxq.fos.b(PitayaSkillDetailFragment.kt:499)
	at ryxq.fos.b(PitayaSkillDetailFragment.kt:367)
	at ryxq.fos.a(PitayaSkillDetailFragment.kt:293)
	at ryxq.fot$c.a(PitayaSkillDetailPresenter.kt:138)
	at ryxq.fot$c.accept(PitayaSkillDetailPresenter.kt:61)
	at io.reactivex.internal.observers.ConsumerSingleObserver.a_(ConsumerSingleObserver.java:62)
	at io.reactivex.internal.operators.single.SingleObserveOn$ObserveOnSingleObserver.run(SingleObserveOn.java:81)
	at ryxq.gyn$b.run(HandlerScheduler.java:124)
	at android.os.Handler.handleCallback(Handler.java:742)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:157)
	at android.app.ActivityThread.main(ActivityThread.java:5653)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:746)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:636)
```

Code:
```Kotlin
val controllerListener = object: BaseControllerListener<ImageInfo>(){

            override fun onFinalImageSet(id: String?, imageInfo: ImageInfo?, animatable: Animatable?) {
                super.onFinalImageSet(id, imageInfo, animatable)
                imageInfo?.let {
                    //.....
                        if (ih-height > radius || iw-width>radius) {
                            mIvSkill?.hierarchy?.roundingParams = null
                            mIvSkill?.hierarchy?.setBackgroundImage(resources?.getDrawable(R.drawable.shape_accompany_master_skill))
                        } else {
                            mIvSkill?.hierarchy?.roundingParams = RoundingParams.fromCornersRadius(6.dpToPx.toFloat())
                        }
                }
            }
}
val controller = Fresco.newDraweeControllerBuilder()
            .setControllerListener(controllerListener)
            .setUri(url)
            .build()

mIvSkill?.hierarchy?.setPlaceholderImage(R.drawable.shape_bg_eee_radius_6)
//Crash while run to this line!!!!!
mIvSkill?.hierarchy?.setFailureImage(R.drawable.ic_skill_detail_license, ScalingUtils.ScaleType.FIT_XY)
mIvSkill?.hierarchy?.setActualImageScaleType(ScalingUtils.ScaleType.FIT_CENTER)
mIvSkill?.controller = controller
```
- Devices: Xiaomi MI NOTE LTE
- Android Version: 6.0.1
